app/views/_user_util_links.html.erb

```diff
@@ -0,0 +1,31 @@
+<div class="navbar-right">
+  <ul class="nav navbar-nav">
+    <%= render_nav_actions do |config, action|%>
+      <li><%= action %></li>
+    <% end %>
+  </ul>
+
+  <% if has_user_authentication_provider? %>
+    <ul class="nav navbar-nav">
+      <% if current_user %>
+        <% if can? :create, GenericAsset %>
```

Comment:
I'd like to see the "is admin" functionality centralized - this is the same check as in the controller, but if we change the definition of an admin, we have to fix it twice.